test/src/userobjects/ElemSideNeighborLayersVarTester.C

```diff
@@ -0,0 +1,109 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org All rights reserved, see COPYRIGHT
+//* for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT Licensed
+//* under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ElemSideNeighborLayersVarTester.h"
+
+#include "libmesh/parallel.h"
+
+registerMooseObject("MooseTestApp", ElemSideNeighborLayersVarTester);
+
+template <>
+InputParameters
+validParams<ElemSideNeighborLayersVarTester>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+  params.addClassDescription(
+      "Tests whether a coupled variable is correctly ghosted.  This code needs to be cleaned up");
+  params.addRequiredParam<unsigned int>("rank", "The rank for which the ghosted data are recorded");
+
+  params.registerRelationshipManagers("ElementSideNeighborLayers");
+  params.addRequiredParam<unsigned short>("element_side_neighbor_layers",
+                                          "Number of layers to ghost");
```

Comment:
Is it a good idea to have a default value such as 1

Reply:
Sure, i can make a default value.  This class we really just designed to show the "bug" (if one exists) and not to be super wonderful